daemon/response.go

```diff
@@ -100,12 +94,8 @@ func (r *resp) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
 
 	hdr := w.Header()
 	if r.Status == http.StatusAccepted || r.Status == http.StatusCreated {
-		if m, ok := r.Result.(map[string]interface{}); ok {
-			if location, ok := m["resource"]; ok {
-				if location, ok := location.(string); ok && location != "" {
-					hdr.Set("Location", location)
-				}
-			}
+		if r.Meta.Change != "" {
+			hdr.Set("Location", changeURL(r.Meta.Change))
```

Comment:
Is that really sound in HTTP terms? Feels like this should be the URL of the thing created (e.g.) instead of the change that creates it.